File: API/Entities/OrderAggregate/Order.cs
using System;
using System.Collections.Generic;
using API.Entities.OrderAggregate;

namespace API.Entities.OrderAggregate
{
    /**
     * C# model for an order,
     * This represents a customer order in the system
     */
    public class Order
    {
        public int Id { get; set; }
        public required string BuyerEmail { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public List<OrderItem> OrderItems { get; set; } = [];
        public decimal Subtotal { get; set; }
        public decimal DeliveryFee { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.Pending;
        public required ShippingAddress ShippingAddress { get; set; }
        public required PaymentSummary PaymentSummary { get; set; }
        public required string PaymentIntentId { get; set; }
        public long Discount { get; set; }


        public decimal GetTotal()
        {
            return Subtotal + DeliveryFee - Discount;
        }
    }
}

File: client/src/app/models/order.ts
export interface OrderItem {
  productId: number;
  name: string;
  pictureUrl: string;
  price: number;
  quantity: number;
}

export interface ShippingAddress {
  name: string;
  line1: string;
  line2?: string | null;
  city: string;
  state: string;
  postal_code: string;
  country: string;
}

export interface PaymentSummary {
  last4: number;
  brand: string;
  expMonth: number;
  expYear: number;
}

export interface Order {
  id: number;
  buyerEmail: string;
  orderDate: string;
  orderItems: OrderItem[];
  subtotal: number;
  deliveryFee: number;
  status: string;
  shippingAddress: ShippingAddress;
  paymentSummary: PaymentSummary;
  total: number;
  discount: number;
}

export interface CreateOrder {
  shippingAddress: ShippingAddress;
  paymentSummary: PaymentSummary;
}
